src/python/doc/wasserstein_distance_user.rst

```diff
@@ -58,16 +58,29 @@ An index of -1 represents the diagonal.
     import gudhi.wasserstein
     import numpy as np
 
-    diag1 = np.array([[2.7, 3.7],[9.6, 14.],[34.2, 34.974]])
-    diag2 = np.array([[2.8, 4.45], [5, 6], [9.5, 14.1]])
-    cost, matching = gudhi.wasserstein.wasserstein_distance(diag1, diag2, matching=True, order=1., internal_p=2.)
-
-    message = "Wasserstein distance value = %.2f, optimal matching: %s"  %(cost, matching)
-    print(message)
+    dgm1 = np.array([[2.7, 3.7],[9.6, 14.],[34.2, 34.974]])
+    dgm2 = np.array([[2.8, 4.45], [5, 6], [9.5, 14.1]])
+    cost, matchings = gudhi.wasserstein.wasserstein_distance(diag1, diag2, matching=True, order=1., internal_p=2.)
+
+    message_cost = "Wasserstein distance value = %.2f" %cost
+    print(message_cost)
+    dgm1_to_diagonal = matchings[np.where(matchings[:,0] == -1)][:,1]
```

Comment:
I don't think you need `np.where`, `a[(a[:,0]==-1)][:,1]` or even simpler `a[a[:,0]==-1,1]` should do, no? Or `np.compress(a[:,0]==-1,a[:,1])` or any of many variations.

Reply:
I think so. I will investigate asap.